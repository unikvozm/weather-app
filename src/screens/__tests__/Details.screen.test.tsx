import React from 'react';
import renderer from 'react-test-renderer';
import {Provider as PaperProvider} from 'react-native-paper';
import {Provider as ReduxProvider} from 'react-redux';
import createSagaMiddleware from 'redux-saga';
import {createStore, applyMiddleware} from 'redux';
import {DetailsScreen} from '../Details.screen';
import {rootReducer} from '../../redux/reducers/root.reducer';

const saga = createSagaMiddleware();
const store = createStore(rootReducer, applyMiddleware(saga));

test('DetailsScreen renders correctly', () => {
  const tree = renderer
    .create(
      <ReduxProvider store={store}>
        <PaperProvider>
          <DetailsScreen
            route={{params: {cityId: 1}, key: 'string', name: 'Details'}}
          />
        </PaperProvider>
      </ReduxProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
