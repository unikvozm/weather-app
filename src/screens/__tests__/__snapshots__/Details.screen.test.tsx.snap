// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text>
      Something went wrong
    </Text>
  </View>
  ,
</View>
`;
